3p/ampcontext.js

```diff
@@ -210,16 +217,43 @@ export class AbstractAmpContext {
    *  @param {number} width The new width for the ad we are requesting.
    *  @param {number} height The new height for the ad we are requesting.
    *  @param {boolean=} hasOverflow Whether the ad handles its own overflow ele
+   *  @return {Promise} Signify the success/failure of the request.
    */
   requestResize(width, height, hasOverflow) {
+    const requestId = this.nextResizeRequestId_++;
     this.client_.sendMessage(
       MessageType.EMBED_SIZE,
       dict({
+        'id': requestId,
         'width': width,
         'height': height,
         'hasOverflow': hasOverflow,
       })
     );
+    return new Promise((resolve, reject) => {
+      this.resizeIdToHandlers_[requestId] = [resolve, reject];
+    });
+  }
+
+  /**
+   *  Set up listeners to handle responses from request size.
+   */
+  listenToResizeResponse_() {
+    this.client_.registerCallback(MessageType.EMBED_SIZE_CHANGED, (data) => {
+      const id = data['id'];
+      if (id !== undefined) {
+        this.resizeIdToHandlers_[id][0]();
```

Comment:
nit: maybe name these so it's easier to understand what is happening. eg. `this.resizeIdToHandlers_[id].resolve()`